bug: Failing rfork tests

I'm getting this:

```
--- FAIL: TestExecuteRforkUserNS (0.00s)
        shell_linux_test.go:72: fork/exec /home/katz/workspace/go/src/github.com/NeowayLabs/nash/cmd/nash/nash: operation not permitted
=== RUN   TestExecuteRforkEnvVars
--- FAIL: TestExecuteRforkEnvVars (0.00s)
        shell_linux_test.go:107: fork/exec /home/katz/workspace/go/src/github.com/NeowayLabs/nash/cmd/nash/nash: operation not permitted
=== RUN   TestExecuteRforkUserNSNested
--- FAIL: TestExecuteRforkUserNSNested (0.00s)
        shell_linux_test.go:142: fork/exec /home/katz/workspace/go/src/github.com/NeowayLabs/nash/cmd/nash/nash: operation not permitted
=== RUN   TestExecuteIssue68
```

The same code runs on the CI with no problem and it seems to have started since I updated my arch linux x_x, I would guess that something changed on the kernel:

```
(katz)λ> uname -a
Linux archlinux 4.14.9-1-ARCH #1 SMP PREEMPT Tue Dec 26 00:18:37 UTC 2017 x86_64 GNU/Linux
```

--- a/internal/sh/shell_linux_test.go
+++ b/internal/sh/shell_linux_test.go
@@ -37,7 +37,7 @@ func init() {
 	val, permerr := ioutil.ReadFile(usernsCfg)
 
 	// Travis build doesn't support /proc/config.gz but kernel has userns
-	if os.Getenv("TRAVIS_BUILD") == usernsOk {
+	if os.Getenv("TRAVIS_BUILD") == "1" {
 		enableUserNS = permerr == nil && string(val) == usernsOk
 		logUsernsDetection(permerr)
 		return

--- a/internal/sh/shell_linux_test.go
+++ b/internal/sh/shell_linux_test.go
@@ -49,7 +49,7 @@ func init() {
 		return
 	}
 
-	// old kernels dont has sysctl configurations
+	// old kernels dont have sysctl configurations
 	// than just checking the /proc/config suffices
 	usernsCmd := exec.Command("zgrep", kernelcfg, "/proc/config.gz")
 

--- a/internal/sh/shell_linux_test.go
+++ b/internal/sh/shell_linux_test.go
@@ -23,11 +23,13 @@ func init() {
 	logUsernsDetection := func(err error) {
 		if enableUserNS {
 			fmt.Printf("Linux user namespaces enabled!")
-		} else {
-			fmt.Printf("ERROR: %s\n", err.Error())
-			fmt.Printf("Warning: Impossible to know if kernel support USER namespace.\n")
-			fmt.Printf("Warning: USER namespace tests will not run.\n")
 			return
+		}
+
+		fmt.Printf("Warning: Impossible to know if kernel support USER namespace.\n")
+		fmt.Printf("Warning: USER namespace tests will not run.\n")
+		if err != nil {
+			fmt.Printf("ERROR: %s\n", err)
 		}
 	}
 

--- a/internal/sh/shell_linux_test.go
+++ b/internal/sh/shell_linux_test.go
@@ -5,6 +5,7 @@ package sh
 import (
 	"bytes"
 	"fmt"
+	"io/ioutil"
 	"os"
 	"os/exec"
 	"strings"
@@ -16,30 +17,50 @@ var (
 )
 
 func init() {
-	// Travis build doesn't support /proc/config.gz but have userns enabled
-	if os.Getenv("TRAVIS_BUILD") == "1" {
-		enableUserNS = true
+	const usernsOk = "1"
+	const kernelcfg = "CONFIG_USER_NS"
 
+	logUsernsDetection := func(err error) {
+		if enableUserNS {
+			fmt.Printf("Linux user namespaces enabled!")
+		} else {
+			fmt.Printf("ERROR: %s\n", err.Error())
+			fmt.Printf("Warning: Impossible to know if kernel support USER namespace.\n")
+			fmt.Printf("Warning: USER namespace tests will not run.\n")
+			return
+		}
+	}
+
+	usernsCfg := "/proc/sys/kernel/unprivileged_userns_clone"
+	val, permerr := ioutil.ReadFile(usernsCfg)
+
+	// Travis build doesn't support /proc/config.gz but kernel has userns
+	if os.Getenv("TRAVIS_BUILD") == usernsOk {
+		enableUserNS = permerr == nil && string(val) == usernsOk
+		logUsernsDetection(permerr)
 		return
 	}
 
-	usernsCmd := exec.Command("zgrep", "CONFIG_USER_NS", "/proc/config.gz")
+	if permerr == nil {
+		enableUserNS = string(val) == usernsOk
+		logUsernsDetection(permerr)
+		return
+	}
+
+	// old kernels dont has sysctl configurations
+	// than just checking the /proc/config suffices
+	usernsCmd := exec.Command("zgrep", kernelcfg, "/proc/config.gz")
 
 	content, err := usernsCmd.CombinedOutput()
-
 	if err != nil {
-		fmt.Printf("ERROR: %s\n", err.Error())
-		fmt.Printf("Warning: Impossible to know if kernel support USER namespace.\n")
-		fmt.Printf("Warning: USER namespace tests will not run.\n")
 		enableUserNS = false
+		logUsernsDetection(fmt.Errorf("Failed to get kernel config: %s", err))
+		return
 	}
 
-	switch strings.Trim(string(content), "\n \t") {
-	case "CONFIG_USER_NS=y":
-		enableUserNS = true
-	default:
-		enableUserNS = false
-	}
+	cfgVal := strings.Trim(string(content), "\n\t ")
+	enableUserNS = cfgVal == kernelcfg+"=y"
+	logUsernsDetection(fmt.Errorf("%s not enabled in kernel config", kernelcfg))
 }
 
 func TestExecuteRforkUserNS(t *testing.T) {

